app/Http/Controllers/Api/FleetController.php

```diff
@@ -69,4 +69,38 @@ public function get_aircraft($id, Request $request)
 
         return new AircraftResource($aircraft);
     }
+
+    /**
+     * Get aircrafts from actual airport.
+     * State, status and rank are optional
+     * /api/aircraft/airport/ICAO?state=0&status=B&rank=idrank
+     *
+     * @param         $id
+     * @param Request $request
+     *
+     * @return AircraftResource
+     */
+    public function get_airport($id, Request $request)
+    {
+        $aircrafts = $this->aircraftRepo
+            ->with(['subfleet', 'subfleet.fares', 'subfleet.ranks']);
+
+        if ($request->filled('rank')) {
```

Comment:
Is this to determine for a user's rank, or any specific rank?

Reply:
This allow any idranks, because it's possible that aircraft are not restricted for ranks, or that only need show statics or other data about airports.